MSW/MSW/Program.cs

```diff
@@ -36,12 +40,19 @@ static void Main(string[] a_args)
 			{
 				stopwatch.Restart();
 				watchdog.Tick();
+				if (m_tickCountDown == 0)
+				{
+					m_tickCountDown = ServerToLocalTickTimeMsFactor;
+					watchdog.ServerTick();
+				}
 
 				long timeToSleep = TickTimeMs - stopwatch.ElapsedMilliseconds;
 				if (timeToSleep > 0)
 				{
 					Thread.Sleep((int)timeToSleep);
 				}
+
+				m_tickCountDown--;
```

Comment:
This approach assumes a constant tick frequency, is that reliable?

Reply:
As long as `stopwatch.ElapsedMilliseconds` is less than TickTimeMs (50ms) it will be fine since `timeToSleep` will be larger than 0 , and it will go to sleep to make sure the local tick will endure for the 50ms.
However, if `stopwatch.ElapsedMilliseconds` is larger than 50 ms for some reason, `timeToSleep` will be negative, and in that case the `tickCountDown` will only be decreased once, although more time was consumed.

I will rewrite it using 2 stopwatches, adding a bit more complexity to the sleep